doc/rtd/topics/configuration.rst

```diff
@@ -0,0 +1,89 @@
+.. _configuration:
+
+Configuration Sources
+*********************
+
+Internally, cloud-init builds a single configuration that is then referenced
+throughout the life of cloud-init. The configuration is built from multiple
+sources such that if a key is defined in multiple sources, the higher priority
+source overwrites the lower priority source.
+
+Base Configuration
+==================
+
+From lowest priority to highest, configuration sources are:
+
+* **Hardcoded config**: Config_ that lives within the source of cloud-init
+  and cannot be changed.
+* **Configuration directory**: Anything defined in ``/etc/cloud/cloud.cfg`` and
+  ``/etc/cloud/cloud.cfg.d``.
+* **Runtime config**: Anything defined in ``/run/cloud-init/cloud.cfg``.
+* **Kernel command line**: On the kernel command line, anything found between
+  ``cc:`` and ``end_cc`` will be interpreted as cloud-config user data.
+
+These four sources make up the base configuration.
+
+Vendor and User Data
+====================
+Added to the base configuration are:
+
+* **Vendor data**: :ref:`Data<vendordata>` provided by the datasource
+* **User data**: :ref:`Data<user_data_formats>` also provided by
+  the datasource
+
+These get fetched from the datasource and are defined at instance launch.
+
+.. note::
+  While much of what is defined in the base configuration can be overridden by
+  vendor data and user data, base configuration sources do not conform to
+  :ref:`#cloud-config<topics/format:Cloud Config Data>`
+
+Network Configuration
+=====================
+Network configuration happens independently from other cloud-init
+configuration. It is sourced from the cloud provided metadata service and is
+not user editable. If you need to change your network configuration,
+your options are:
+
+* Use the cloud-provided tools to update your network topology in the cloud.
+* :ref:`Disable cloud-init network configuration entirely
```

Comment:
A third option here is to provide network configuration in a customized image directly in `/etc/cloud/cloud.cfg.d/*.cfg` with a network: <<config>> definition.